utilities/xmlrpcpp/src/XmlRpcClient.cpp

```diff
@@ -506,15 +507,31 @@ XmlRpcClient::parseResponse(XmlRpcValue& result)
     if ( ! result.fromXml(_response, &offset)) {
       XmlRpcUtil::error("Error in XmlRpcClient::parseResponse: Invalid response value. Response:\n%s", _response.c_str());
       _response = "";
+      _header = "";
       return false;
     }
   } else {
     XmlRpcUtil::error("Error in XmlRpcClient::parseResponse: Invalid response - no param or fault tag. Response:\n%s", _response.c_str());
     _response = "";
+    _header = "";
     return false;
   }
       
   _response = "";
+
+  if (_header.size() < 8 || _header.find("HTTP/") != 0)
+    XmlRpcUtil::error("Error XmlRpcClient::parseResponse: Header does not start with protocol");
```

Comment:
In case a response is received which doesn't contain the header this patch will add a new error. Is this really necessary?

I am also not sure about if it is reasonable to change the current behavior for HTTP 1.0 connections to not-keep-them-open.

Also how does this related to #1287?

Reply:
It's not really necessary, I can remove it.

Reply:
@mgrrx This comment seems to be still pending.

Reply:
sure will do ASAP

Reply:
I simplified the state ment. I assume it does the job without introducing the new error message.

Reply:
> I assume it does the job without introducing the new error message.

Please make sure to test the latest change since this is a significant change for a distro which is already stable for 4 years.